test/testutil/tap_bio.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2017, Oracle and/or its affiliates.  All rights reserved.
```

Comment:
I'm from Oracle.  The open source contribution policy requires me to include this.